tests/viewer/test_event_db.cpp

```diff
@@ -0,0 +1,147 @@
+{
+    // Arrange
+    Query query;
+    HT_TimestampNs start_ts = (HT_TimestampNs)100;
+    HT_TimestampNs stop_ts = (HT_TimestampNs)200;
+
+    // Act
+    event_db.insert(*_event2);
+    event_db.insert(*_event1);
+    
+    // Assert
+    auto response = event_db.get_data(start_ts, stop_ts, query);
+    std::vector<Event> correct_response = {*_event1, *_event2};
+
+    ASSERT_EQ(response.size(), correct_response.size());
+    for (unsigned int i = 0; i < response.size(); ++i)
+    {
+        ASSERT_EQ(response[i].get().get_timestamp(), correct_response[i].get_timestamp());
+    }
+}
+
+TEST_F(TestEventDB, QuerySpecifcKlassId)
+{
+    // Arrange
+    Query query;
+    HT_TimestampNs start_ts = (HT_TimestampNs)100;
+    HT_TimestampNs stop_ts = (HT_TimestampNs)300;
+    query.klass_id = 1;
+
+    // Act
+    event_db.insert(*_event1);
+    event_db.insert(*_event2);
+    event_db.insert(*_event3);
+    auto response = event_db.get_data(start_ts, stop_ts, query);
+
+    // Assert
+    std::vector<Event> correct_response = {*_event1, *_event3};
+    ASSERT_EQ(response.size(), correct_response.size());
+    for (unsigned int i = 0; i < response.size(); ++i)
+    {
+        ASSERT_EQ(response[i].get().get_timestamp(), correct_response[i].get_timestamp());
+    }
+}
+
+
+TEST_F(TestEventDB, CheckLowerBoundConditions)
+{
+    // Arrange
+    Query query1, query2, query3;
+    HT_TimestampNs start_ts;
+    HT_TimestampNs stop_ts;
+
+    // Act
+    event_db.insert(*_event1);
+    event_db.insert(*_event2);
+    event_db.insert(*_event3);
+    event_db.insert(*_event4);
+
+    start_ts = (HT_TimestampNs)100;
+    stop_ts = (HT_TimestampNs)400;
+    auto response1 = event_db.get_data(start_ts, stop_ts, query1);
+
+    start_ts = (HT_TimestampNs)0;
+    stop_ts = (HT_TimestampNs)500;
+    auto response2 = event_db.get_data(start_ts, stop_ts, query2);
+
+    start_ts = (HT_TimestampNs)150;
+    stop_ts = (HT_TimestampNs)350;
+    auto response3 = event_db.get_data(start_ts, stop_ts, query3);
+
+    // Assert
+    std::vector<Event> correct_response1 = {*_event1, *_event2, *_event3, *_event4};
+    std::vector<Event> correct_response2 = {*_event1, *_event2, *_event3, *_event4};
+    std::vector<Event> correct_response3 = {*_event2, *_event3};
+    ASSERT_EQ(response1.size(), correct_response1.size());
+    for (unsigned int i = 0; i < response1.size(); ++i)
+    {
+        ASSERT_EQ(response1[i].get().get_timestamp(), correct_response1[i].get_timestamp());
+    }
+    ASSERT_EQ(response2.size(), correct_response2.size());
+    for (unsigned int i = 0; i < response2.size(); ++i)
+    {
+        ASSERT_EQ(response2[i].get().get_timestamp(), correct_response2[i].get_timestamp());
+    }
+    ASSERT_EQ(response3.size(), correct_response3.size());
+    for (unsigned int i = 0; i < response3.size(); ++i)
+    {
+        ASSERT_EQ(response3[i].get().get_timestamp(), correct_response3[i].get_timestamp());
+    }
+}
+
```

Comment:
I think we need to have a few more tests:
 * Could you add test where `start_ts > stop_ts`?
 * Could you add test where database is empty?
 * Could you add test where `start_ts` == `stop_ts` and one of the events' timestamp is equal to `start_ts`?